core/eolearn/core/eoexecution.py

```diff
@@ -58,59 +61,72 @@ class EOExecutor:
+        self.execution_kwargs = self._parse_and_validate_execution_kwargs(execution_kwargs)
+        self.execution_names = self._parse_execution_names(execution_names, self.execution_kwargs)
         self.save_logs = save_logs
-        self.logs_folder = os.path.abspath(logs_folder)
+        self.filesystem, self.logs_folder = self._parse_logs_filesystem(filesystem, logs_folder)
         self.logs_filter = logs_filter
-        self.filesystem = filesystem
+        self.logs_handler_factory = logs_handler_factory
 
         self.start_time = None
         self.report_folder = None
         self.general_stats = {}
         self.execution_results = None
 
     @staticmethod
-    def _parse_and_validate_execution_args(
-            execution_args: Sequence[Dict[EONode, Dict[str, object]]]) -> List[Dict[EONode, Dict[str, object]]]:
+    def _parse_and_validate_execution_kwargs(
+            execution_kwargs: Sequence[Dict[EONode, Dict[str, object]]]) -> List[Dict[EONode, Dict[str, object]]]:
         """ Parses and validates execution arguments provided by user and raises an error if something is wrong
         """
-        if not isinstance(execution_args, (list, tuple)):
-            raise ValueError("Parameter 'execution_args' should be a list")
+        if not isinstance(execution_kwargs, (list, tuple)):
+            raise ValueError("Parameter 'execution_kwargs' should be a list")
 
-        for input_kwargs in execution_args:
+        for input_kwargs in execution_kwargs:
             EOWorkflow.validate_input_kwargs(input_kwargs)
 
-        return [input_kwargs or {} for input_kwargs in execution_args]
+        return [input_kwargs or {} for input_kwargs in execution_kwargs]
 
     @staticmethod
-    def _parse_execution_names(execution_names: Optional[List[str]], execution_args: Sequence) -> List[str]:
+    def _parse_execution_names(execution_names: Optional[List[str]], execution_kwargs: Sequence) -> List[str]:
         """ Parses a list of execution names
         """
         if execution_names is None:
-            return [str(num) for num in range(1, len(execution_args) + 1)]
+            return [str(num) for num in range(1, len(execution_kwargs) + 1)]
 
-        if not isinstance(execution_names, (list, tuple)) or len(execution_names) != len(execution_args):
+        if not isinstance(execution_names, (list, tuple)) or len(execution_names) != len(execution_kwargs):
             raise ValueError("Parameter 'execution_names' has to be a list of the same size as the list of "
                              "execution arguments")
         return execution_names
 
+    @staticmethod
+    def _parse_logs_filesystem(filesystem: Optional[fs.base.FS], logs_folder: str) -> Tuple[fs.base.FS, str]:
+        """ Ensures a filesystem and a file path relative to it.
+        """
+        if filesystem is None:
+            return get_base_filesystem_and_path(logs_folder)
+        return filesystem, logs_folder
```

Comment:
So if filesystem is `None` but `logs_folder` is something, the path gets ignored? Is this the desired behaviour?

Reply:
No, In that case we assume `logs_folder` is an absolute path, hence `get_base_filesystem_and_path` will split it into a filesystem object and a path relative to the object.

Reply:
ah, i missed the part where it is passed to `get_base_filesystem_and_path`